src/Features/Core/Portable/GenerateConstructorFromMembers/AbstractGenerateConstructorFromMembersCodeRefactoringProvider.cs

```diff
@@ -47,60 +49,149 @@ protected AbstractGenerateConstructorFromMembersCodeRefactoringProvider() : this
+            Document currentDocument,
+            string? serializedIntentData,
+            CancellationToken cancellationToken)
+        {
+            using var _ = ArrayBuilder<CodeAction>.GetInstance(out var actions);
+            await ComputeRefactoringsAsync(
+                priorDocument,
+                priorSelection,
+                (singleAction, applicableToSpan) => actions.Add(singleAction),
+                (multipleActions) => actions.AddRange(multipleActions),
+                cancellationToken).ConfigureAwait(false);
+
+            if (actions.IsEmpty())
+            {
+                return ImmutableArray<IntentProcessorResult>.Empty;
+            }
+
+            // The refactorings returned will be in the following order (if available)
+            // FieldDelegatingCodeAction, ConstructorDelegatingCodeAction, GenerateConstructorWithDialogCodeAction
+            using var resultsBuilder = ArrayBuilder<IntentProcessorResult>.GetInstance(out var results);
+            foreach (var action in actions)
+            {
+                var intentResult = await GetIntentProcessorResultAsync(action, cancellationToken).ConfigureAwait(false);
+                results.AddIfNotNull(intentResult);
+            }
+
+            return results.ToImmutable();
+
+            static async Task<IntentProcessorResult?> GetIntentProcessorResultAsync(CodeAction codeAction, CancellationToken cancellationToken)
+            {
+                var title = codeAction.Title;
+                var operations = await GetCodeActionOperationsAsync(codeAction, cancellationToken).ConfigureAwait(false);
+
+                // Generate ctor will only return an ApplyChangesOperation or potentially document navigation actions.
+                // We can only return edits, so we only care about the ApplyChangesOperation.
+                var applyChangesOperation = operations.OfType<ApplyChangesOperation>().SingleOrDefault();
+                if (applyChangesOperation == null)
+                {
+                    return null;
+                }
+
+                var type = typeof(CodeAction);
+                return new IntentProcessorResult(applyChangesOperation.ChangedSolution, title, type.Name);
+            }
+
+            static async Task<ImmutableArray<CodeActionOperation>> GetCodeActionOperationsAsync(
+                CodeAction action,
+                CancellationToken cancellationToken)
+            {
+                if (action is CodeActionWithOptions)
+                {
+                    // Generate ctor only returns GenerateConstructorWithDialogCodeAction for CodeActionWithOptions
+                    var dialogAction = (GenerateConstructorWithDialogCodeAction)action;
+
+                    // Usually applying this code action pops up a dialog allowing the user to choose which options.
+                    // We can't do that here, so instead we just take the defaults until we have more intent data.
```

Comment:
that's cool.  woudl love to see a demo where this works.  i.e. where they pass in "we think tehy are generating teh constructor with props X and Y".